### DISCUSSION TITLE
Fast resize and move

### DISCUSSION THREAD

**Discussion by Ryan7865** [labwc/labwc#1037] [Q&A]
I find glitches when moving or resizing terminal windows quickly (the way I am used to on X11 / Fvwm).

Resizing ("W-Right" + "Drag" bound to "Resize"):
After several resize moves, the prompt, or fractions of it, shows up repeatedly.
This happens quite often with urxvt (which I am not ready to give up on just yet), but also with foot and alacritty.
Here is a screenshot of "foot":
![image](https://github.com/labwc/labwc/assets/142569464/e04df798-5582-47d4-975f-a774f7264ca6)

Presumably, this is really the application's fault, but it would not happen if its contents need not be continuously updated. Fvwm has the option to hide the contents while resizing, showing only a grid until the mouse button is released.

Moving ("W-Left" + "Drag" bound to "Move"):
While pressing the modifier key and moving the mouse over the window, I click the left mouse button. If this happens fast, sometimes text is selected (urxvt, alacritty) instead of the window being moved. It seems to be a race condition. I haven't seen text selected in foot, but instead the window just "slips through" and doesn't move. Perhaps the mouse button is released too quickly. Anyway, this never happens when everything is done slowly (press mouse button, wait a bit, then move the mouse).

**Comment by Consolatis** [comment]
> After several resize moves, the prompt, or fractions of it, shows up repeatedly.
> ..
> Presumably, this is really the application's fault [..]

Agree, this is most definitely an application bug (and I am pretty sure that I've seen this since basically the first time I used any terminal emulator on X11). So there is nothing we can do here other than asking terminal emulator devs to fix it.

> [..] but it would not happen if its contents need not be continuously updated.

That is possible but we prefer not working around client bugs.

However, while implementing the `<resize>` config section because of the optional position / size overlay I've stumbled over the `drawContents` setting in the [Openbox wiki](http://openbox.org/wiki/Help:Configuration#Resize_.28and_move.29). It is something that would make sense to implement to cope with very low resource systems (or potentially even when running with software rendering). So to sum up: I am all for implementing a `delay-resize-until-finished` setting but for other reasons than stated here. There is no time frame for it as we would have to investigate how invasive these changes would actually be. The whole section about `<resize>` at https://github.com/labwc/labwc-scope#26-configuration---resize-and-move is missing the arguments so it seems the labwc project as a whole has not yet decided if it wants to implement that setting or not. Personally I'd absolutely vote for a yes here.

> While pressing the modifier key and moving the mouse over the window, I click the left mouse button. If this happens fast, sometimes text is selected (urxvt, alacritty) instead of the window being moved. It seems to be a race condition

Interesting. It would be worth to investigate this further to figure out the exact conditions when this happens. Maybe the difference is that the mouse button down event is detected first before the modifier key is recognized?

**Reply by johanmalm** [reply]
I support both `drawContents` and `delay-resize-until-finished` (approach to-be-worked-out)

**Comment by Ryan7865** [comment]
Many thanks for considering this!

Besides terminal emulators, I found other applications that have trouble in keeping up during the resize, albeit without lasting effect. One is Vivaldi, which I am using right now. The lag is quite obvious in XWayland and much more subtle if Wayland is used directly, but present. I would consider my hardware to be quite good.

Concerning the "move" problem, I have created a screenshot of Gimp that perhaps makes it more obvious:
![image](https://github.com/labwc/labwc/assets/142569464/792f9833-b6bc-42df-b1b3-a664150142e4)
The modifier key was pressed all the time while drawing this. I don't think there is a special trick to it besides doing it fast. Most times the window moved as it should, at all other times Gimp drew a line.
